### DISCUSSION TITLE
no Route matched with those values

### DISCUSSION THREAD

**Discussion by mohsen-abedy** [Kong/kong#12732] [Help] [answered]
Can you help me with how to resolve the following error for the below configuration?
kong version: 3.6
{ "protocols": [ "http", "https" ], "sources": null, "destinations": null, "created_at": 1710410167, "updated_at": 1710410372, "regex_priority": 0, "snis": null, "service": { "id": "eab847e9-94d8-409c-bd83-6ed6057cdc22" }, "paths": [ "/partner/api/v1/amenities/(?<amenityId>\S+)" ], "request_buffering": true, "response_buffering": true, "tags": [ "amenities-partner" ], "strip_path": false, "hosts": null, "https_redirect_status_code": 426, "name": "GET-partner-api-v1-amenities-amenityId", "methods": [ "GET" ], "headers": null, "path_handling": "v0", "preserve_host": false, "id": "4e4437d9-22fb-474e-9b86-be5889d3e778" }

**Comment by chronolaw** [comment]
Could you give us the re-produce steps in details? thanks.

**Reply by mohsen-abedy** [reply]
@chronolaw 

There is my service config:

`{
  "name": "core",
  "tags": [
    "core"
  ],
  "protocol": "http",
  "path": null,
  "read_timeout": 60000,
  "retries": 5,
  "host": "192.168.1.104",
  "connect_timeout": 60000,
  "tls_verify_value": false,
  "tls_verify_enabled": false,
  "ca_certificates": null,
  "client_certificate": null,
  "write_timeout": 60000,
  "port": 3002
}`

After created a service I created a route with this config:

`{ "protocols": [ "http", "https" ], "sources": null, "destinations": null, "created_at": 1710410167, "updated_at": 1710410372, "regex_priority": 0, "snis": null, "service": { "id": "eab847e9-94d8-409c-bd83-6ed6057cdc22" }, "paths": [ "/partner/api/v1/amenities/(?\S+)" ], "request_buffering": true, "response_buffering": true, "tags": [ "amenities-partner" ], "strip_path": false, "hosts": null, "https_redirect_status_code": 426, "name": "GET-partner-api-v1-amenities-amenityId", "methods": [ "GET" ], "headers": null, "path_handling": "v0", "preserve_host": false, "id": "4e4437d9-22fb-474e-9b86-be5889d3e778" }`

It's working in version 2.8 but i am using version 3.6, I get an error 
{
    "message": "no Route matched with those values",
    "request_id": "530c0b037eaae032e9fd0342dab2ba33"
}


I am only experiencing this issue for APIs with URI parameters.
When I call APIs without URI parameters, it works correctly.

**Comment by chronolaw** [comment, marked as answer]
I think that the reason is the field `paths`, since kong 3.0 if it is a regex it must start with `~`.

https://docs.konghq.com/gateway/changelog/#3000:
```
Kong Gateway no longer uses a heuristic to guess whether a route.path is a regex pattern. From 3.0 onward, all regex paths must start with the "~" prefix, and all paths that don’t start with "~" will be considered plain text. The migration process should automatically convert the regex paths when upgrading from 2.x to 3.0. [#9027](https://github.com/Kong/kong/pull/9027)
```
Could you create a new route and try it again?

**Reply by mohsen-abedy** [reply]
Thank you! That works for me.
